server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -98,6 +98,7 @@
     private XContentType contentType;
 
     private String pipeline;
+    private String finalPipeline;
 
     private boolean isPipelineResolved;
```

Comment:
Is the `isPipelineResolved` field still needed and the corresponding if statement in `TransportBulkAction`?
Because before this change, there was validation whether a pipeline on a request was overwriting a required pipeline or in the case that there is both a default and required pipeline. This no longer seems to be the case, and therefor whether a pipeline has been resolved on coordinating node is not really important anymore?

Reply:
I can look into this in a follow-up. My aim is to get the final pipeline behavior in with the minimal change set since this issue is marked as a blocker for an upcoming release.